src/Evaluators/reduced_evaluator.jl

```diff
@@ -119,14 +156,21 @@ initial(nlp::ReducedSpaceEvaluator) = initial(nlp.model, Control())
 bounds(nlp::ReducedSpaceEvaluator, ::Variables) = nlp.u_min, nlp.u_max
 bounds(nlp::ReducedSpaceEvaluator, ::Constraints) = nlp.g_min, nlp.g_max
 
-function update!(nlp::ReducedSpaceEvaluator, u; verbose_level=0)
+function update!(nlp::ReducedSpaceEvaluator, u; verbose_level=0, device = nothing)
     x₀ = nlp.x
     jac_x = nlp.autodiff.Jgₓ
     # Transfer x, u, p into the network cache
     transfer!(nlp.model, nlp.buffer, nlp.x, u, nlp.p)
     # Get corresponding point on the manifold
+    if !(device === nothing)
+        jac_x = AutoDiff.StateJacobian(jac_x, device)
+        nlp  = ReducedSpaceEvaluator(nlp, device)
```

Comment:
Do we know how long does it takes to instantiate on the GPU a new `ReducedSpaceEvaluator` from `nlp`, on the largest instances?

Reply:
For sure. The initialization has to be moved. I'm wondering how to best do this such that one can essentially instantiate the evaluator for a device only if it has not been instantiated before. One evaluator can be on the CPU, CUDA, AMDGPU, or oneAPI.

Reply:
What we could do is to instintiate it for every device type (aka SHADOW COPY), but that may be overkill.